### PR TITLE
Add optimization tests.

### PR DESCRIPTION
**Pull request by floitsch**

### CODE REVIEW

File: tests/optimizations/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,15 @@
+file(GLOB TOIT_OPTIMIZATION_TESTS RELATIVE ${TOIT_SDK_SOURCE_DIR} "*_test.toit")
```
Comment (kasperl) [inline tests/optimizations/CMakeLists.txt:1]: Copyright header.

File: tests/optimizations/byte_array_test.toit [hunk 2]
```diff
@@ -0,0 +1,36 @@
+// Copyright (C) 2021 Toitware ApS. All rights reserved.
```
Comment (kasperl) [inline tests/optimizations/byte_array_test.toit:1]: Again this shows that the tests may need a new 0BSD copyright header.

**Review by kasperl — APPROVED** [review]
Looks good to me.

File: tests/optimizations/CMakeLists.txt (hunk 1 above)
Comment (floitsch) [reply tests/optimizations/CMakeLists.txt:16]: done.

File: tests/optimizations/byte_array_test.toit (hunk 2 above)
Comment (floitsch) [reply tests/optimizations/byte_array_test.toit:1]: I haven't changed the copyrights of any tests yet.
Once the migration is done, I will change them.

**Pull request merged**